processing/src/main/java/org/apache/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -309,7 +325,7 @@ public static SegmentAnalysis mergeAnalyses(
 
     final Map<String, AggregatorFactory> aggregators = new HashMap<>();
 
-    if (lenientAggregatorMerge) {
+    if (AggregatorMergeStrategy.LENIENT == aggregatorMergeStrategy) {
       // Merge each aggregator individually, ignoring nulls
       for (SegmentAnalysis analysis : ImmutableList.of(arg1, arg2)) {
```

Comment:
cant either arg1 and arg2 technically be null here based on logic above? I think that will result in exception here if so.

Reply:
Issue existed before, nvm

Reply:
The existing null checks at the top of the function handle this. If at least one of `arg1` and `arg2` is null, this function returns early:

```java
 if (arg1 == null) {
      return arg2;
    }
 if (arg2 == null) {
      return arg1;
}
```

Reply:
sorry I was confusing arg1, arg2, with id1, id2. nvm. 😆